src/tools/tidy/src/style.rs

```diff
@@ -332,6 +334,11 @@ pub fn check(path: &Path, bad: &mut bool) {
         path.extension().map_or(false, |e| e == "css") && !is_in(path, "src", "librustdoc")
     }
 
+    // This creates a RegexSet as regex contains performance optimizations to be able to deal with these over
+    // 2000 needles efficiently. This runs over the entire source code, so performance matters.
+    let problematic_regex =
+        RegexSetBuilder::new(PROBLEMATIC_CONSTS_STRINGS.as_slice()).case_insensitive(true).build().unwrap();
```

Comment:
isn't just passing the flag enough, why bother with the other check too? i'm also not sure whether it matters enough to do it in the first place, but whatever